clients/src/test/java/org/apache/kafka/clients/producer/internals/DefaultPartitionerTest.java

```diff
@@ -62,4 +67,72 @@ public void testRoundRobinWithUnavailablePartitions() {
+
+    @Test
+    public void testRoundRobin() throws InterruptedException {
+        final String topicA = "topicA";
+        final String topicB = "topicB";
+
+        List<PartitionInfo> allPartitions = asList(new PartitionInfo(topicA, 0, node0, nodes, nodes),
+                new PartitionInfo(topicA, 1, node1, nodes, nodes),
+                new PartitionInfo(topicA, 2, node2, nodes, nodes),
+                new PartitionInfo(topicB, 0, node0, nodes, nodes)
+                );
+        Cluster testCluster = new Cluster("clusterId", asList(node0, node1, node2), allPartitions,
+                Collections.<String>emptySet(), Collections.<String>emptySet());
+
+        final Map<Integer, Integer> partitionCount = new HashMap<>();
+
+        for (int i = 0; i < 30; ++i) {
+            int partition = partitioner.partition(topicA, null, null, null, null, testCluster);
+            Integer count = partitionCount.get(partition);
+            if (null == count) count = 0;
+            partitionCount.put(partition, count + 1);
+
+            if (i % 5 == 0) {
+                partitioner.partition(topicB, null, null, null, null, testCluster);
+            }
+        }
+
+        Assert.assertEquals(10, (int) partitionCount.get(0));
+        Assert.assertEquals(10, (int) partitionCount.get(1));
+        Assert.assertEquals(10, (int) partitionCount.get(2));
+    }
+
+    @Test
+    public void testPartitionPerf() throws Exception {
+        List<PartitionInfo> allPartitions = new ArrayList<PartitionInfo>();
+
+        final int topicCount = 37;
+        for(int i = 0; i < topicCount; ++i) {
+            final String topici = "topic" + i;
+            allPartitions.add(new PartitionInfo(topici, 0, node0, nodes, nodes));
+            allPartitions.add(new PartitionInfo(topici, 1, node1, nodes, nodes));
+            allPartitions.add(new PartitionInfo(topici, 2, node2, nodes, nodes));
+        }
+        final Cluster testCluster = new Cluster("clusterId", asList(node0, node1, node2), allPartitions,
+                Collections.<String>emptySet(), Collections.<String>emptySet());
+
+        int threadCount = 100;
+        final CountDownLatch latch = new CountDownLatch(threadCount);
+        long start = System.currentTimeMillis();
+        final int loopCount = 10000;
+        for(int i = 0; i < threadCount; ++i) {
+            new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    for(int j = 0; j < loopCount; ++j) {
+                        String theTopic = "topic" + j % topicCount;
+                        partitioner.partition(theTopic, null, null, null, null, testCluster);
+                    }
+                    latch.countDown();
+                }
+            }).start();
+        }
+
+        latch.await();
+
+        double avg = (System.currentTimeMillis() - start)/ (double)(threadCount * loopCount);
+        System.out.println("avg partition time(ms)=" + String.format("%5f", avg));
```

Comment:
yeah, since this method is just to get an idea of the performance of Partition.partition method, something like method benchmark